### PR TITLE
Retry Emit for 5xx

### PR DESCRIPTION
**Pull request by willmarks**
## Description

We need to be resilient to 502s. Similar to other retry changes, this PR leverages `axios-retry` to  retry in the case of 5xx's. The PR now constructs the client in `main.ts` and passes it to `emit.ts` via the constructor. This allowed for some separation of concerns when it came to testing the client.

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

File: __tests__/client.test.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      logger,
+      1,
+      100
+    );
+    await expect(client.get('/hi')).rejects.toThrowError(
+      'Request failed with status code 404'
+    );
+    mock.done();
+  });
+});
```
Comment (willmarks) [inline __tests__/client.test.ts:44]: Will add extra line.

**Review by tovbinm — APPROVED** [review]

File: src/main.ts [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
 
+import {makeAxiosInstanceWithRetry} from './client';
 import {Build, Deployment, Emit, Status} from './emit';
 
 const BUILD = 'build';
 const DEPLOYMENT = 'deployment';
 const MODEL_TYPES = [BUILD, DEPLOYMENT];
 
 async function run(): Promise<void> {
+  const logger = pino({name: 'faros-client'});
```
Comment (tovbinm) [inline src/main.ts:12]: you might wanna prettify the first level here

Comment (tovbinm) [reply src/main.ts:12]: e.g. `pino({name: 'faros-client', prettyPrint: {levelFirst: true}}`

Comment (tovbinm) [reply src/main.ts:12]: and that might require another dependency `pino-pretty`

Comment (willmarks) [reply src/main.ts:12]: good call!

**Pull request merged**